pennylane/qnode.py

```diff
@@ -416,7 +416,7 @@ def best_for_op(o_idx):
                 # isn't succeeded by any observables?
                 successors = self._op_successors(o_idx, 'G')
 
-                if all(x.grad_method == 'A' for x in successors):
+                if all(x.supports_heisenberg for x in successors):
```

Comment:
Note: this is quite a massive change, and massively increases the number of CV operations that allow the analytic gradient method. Before, it was assumed that `grad_method='A'` was sufficient, but this is too restrictive.